src/libs/actions/App.js

```diff
@@ -128,6 +121,26 @@ function getPolicyIDList(policies) {
 function openApp(policies) {
```

Comment:
Since `policies` only comes from AuthScreens connecting to the policy collection and passing it to this method, I think it would make more sense to have this action use `Onyx.connect()` to get the policies directly. Otherwise, it's like some kind of indirect way of accessing the policies.

Reply:
In fact, it's already done this way in `reconnectApp()` so they should just be able to both reference the same data.

Reply:
Ah so we discussed `withOnyx` vs `Onyx.connect` in [this thread](https://docs.google.com/document/d/1Mc2fvigSqPnMa6yv6FQk7_opRpjK9kg0e2Gzz9QTXWM/edit?disco=AAAAa73w1KU). It seems like the difference is that in `withOnyx` we don't render the component until the data is read from disk, which makes sure that we have it available before making the request. Using `Onyx.connect` might not guarantee that when `AuthScreen` mounts.

Reply:
We use `Onyx.connect` in ReconnectApp because we only call ReconnectApp from a reconnect callback and not from a component lifecycle method.

Reply:
> It seems like the difference is that in withOnyx we don't render the component until the data is read from disk, which makes sure that we have it available before making the request.

OK, I had to follow a bunch of different thread links, but I understand it now (I ended up [here in slack](https://app.slack.com/client/T03SC9DTT/C02HWMSMZEC/thread/C02HWMSMZEC-1655218680.012429) in the end).

Let's see if we can think a little bit outside-the-box to solve for this. If you use `Onyx.connect()`, then you don't know how many times the `callback` will be triggered to know when you have all the policyIDs. So, you're using a hack of withOnyx and a components lifecycle method to know when all the policyIDs have been read.

So, can we use the same sort of idea with maybe an `Onyx.connectToCollect()` method so that the callback is only triggered once the entire collection has been read from disk? If we can do it with `withOnyx()` then it makes sense we could do it without having to go through a component's lifecycle methods.

Reply:
An `Onyx.connectToCollection` would be nice and certainly avoid this hack. Should we create a separate issue for it and then clean this up once it's implemented?

Reply:
I kind of feel like if we are refactoring things, this is the perfect time to do it. But I don't have a problem with separate PRs as long as it isn't ignored.

Reply:
Cool! I submitted a PR that adds a `waitForCollectionCallback` param to `Onyx.connect` - https://github.com/Expensify/react-native-onyx/pull/163

Reply:
I cleaned this up based on the assumption that we will have a `waitForCollectionCallback` param as mentioned in my comment above! We'll need to update Onyx once that PR is merged!